### ISSUE TITLE
[bug]: [accessibility] Sidebar does not include a focusable dismiss element, trapping mobile screen readers

### ISSUE THREAD

**Issue by 6TELOIV** [shadcn-ui/ui#6761]
### Describe the bug

The sidebar does not (by default) include a button to dismiss the sidebar sheet on mobile.

This has critical accessibility implications for mobile users of screen readers, as once they are in the sidebar drawer, it is impossible to dismiss it (there is no `escape` key to press, and they cannot interact with any element that would cause disimissal).

The only way out I see is a hard refresh.

**My recommended fix:** Add a visible `x` dismissal button to the sidebar, as is present in the sheet component.

### Affected component/components

Sidebar

### How to reproduce

1. Visit a page with a sidebar (ex: https://ui.shadcn.com/view/styles/new-york/sidebar-07)
2. Switch to mobile view (if on desktop)
3. Activate the sidebar
Notice that there is no way to exit the sidebar without pressing escape or clicking the negative space next to the drawer.

On iOS or android screen readers, this makes it impossible to exit the sidebar.

### Codesandbox/StackBlitz link

https://ui.shadcn.com/view/styles/new-york/sidebar-07

### Logs

```bash

```

### System Info

```bash
Android or iOS
Any Browser
```

### Before submitting

- [x] I've made research efforts and searched the documentation
- [x] I've searched for existing issues

**Comment by 6TELOIV** [comment]
Revisiting this now, and the issue is still present.
